[DOC] Add CMake snippet to setup (#3075)
seqan/seqan3: doc/setup/quickstart_cmake/index.md

For example, after adding `another_program.cpp` your `CMakeLists.txt` may look like this:
\snippet test/external_project/seqan3_setup_tutorial/CMakeLists.txt adding_files

# Including SeqAn3 as external project

```cmake
cmake_minimum_required (VERSION 3.14)

project (my_app LANGUAGES CXX VERSION 1.0.0)

> set (seqan3_git_tag "3.2.0")

Reviewer: This will "quickly" become outdated. We should add a note in the release note that we should adapt this here after releasing.

Author: COuld also just be a `# Adapt as needed.`?

Reviewer: That would be better.
```suggestion
set (seqan3_git_tag "#.#.#") # adapt as needed, e.g. "3.2.0"
```

> set (seqan3_git_tag "#.#.#") # adapt as needed, e.g. "3.2.0"

Author: ```suggestion
set (seqan3_git_tag "#.#.#") # adapt as needed, e.g. "3.2.0" or "master"
```

message (STATUS "Fetching SeqAn3 ${seqan3_git_tag}:")

include (FetchContent)
FetchContent_Declare (
    seqan3_fetch_content
    GIT_REPOSITORY "https://github.com/seqan/seqan3.git"
    GIT_TAG "${seqan3_git_tag}"
)

# Download and make SeqAn3 available.
FetchContent_MakeAvailable (seqan3_fetch_content)

add_executable (my_app my_app.cpp)

> # Set up all includes and other dependencies needed by SeqAn3.

Author: ```suggestion
# Set up everything needed to use SeqAn3 with my_app.
```

Reviewer: ```suggestion
# # Set up everything needed to use SeqAn3 with my_app:
```

> target_link_libraries (my_app PUBLIC seqan3::seqan3)

Reviewer: Is the comment line a ToDo or is the comment done with the `target_link_libraries`?

This is not clear to me. Maybe the comment can be rephrased?
